### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by ruicao93**
Update README

Signed-off-by: Rui Cao <rcao@vmware.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,87 @@
...
+
+* The antrea-controller deployment
+* The antrea-agent daemonset
+
+For Openshift 4 clusters, the antrea-operator monitors the network CR in the
+config.openshift.io namespace to update the container network CIDRs used by
+Antrea.
+
+The antrea-operator also monitors a dedicated CR, applies
+changes to antrea-controller and abtrea-agent configuration, and restart the
```
Comment (dantingl) [inline README.md]: abtrea-agent -> antrea-agent
restart -> restarts

File: README.md [hunk 2]
```diff
@@ -1,2 +1,87 @@
...
+The Antrea Operator installs [Antrea](https://github.com/vmware-tanzu/antrea/)
+CNI plugin on an OpenShift Kubernetes cluster.
+The operator will be deployed in the early phases of cluster
+deployment, and it will take care of deploying Antrea components:
+
+* The antrea-controller deployment
+* The antrea-agent daemonset
+
+For Openshift 4 clusters, the antrea-operator monitors the network CR in the
+config.openshift.io namespace to update the container network CIDRs used by
```
Comment (dantingl) [inline README.md]: Do we need to mention that operator also monitors operator crd for the multus config

File: README.md [hunk 3]
```diff
@@ -1,2 +1,87 @@
...
+## Try it out
+
+Generate install-config.yaml by using openshift-install command.
+```
+$ openshift-install --dir=MY_CLUSTER create install-config
+```
+
+Edit `MY_CLUSTER/install-config.yaml` to update networking section.
+Change `networkType` to `antrea`(case insensitive).
+Set container network CIDRs `clusterNetwork` in `MY_CLUSTER/install-config.yaml`.
```
Comment (dantingl) [inline README.md]: antrea does not care about clusterNetwork, how about mentioning service network CIDR here as well

File: README.md [hunk 4]
```diff
@@ -1,2 +1,87 @@
...
+
+Create manifest files:
+```
+$ openshift-install --dir=MY_CLUSTER create manifests
+```
+Put operator yaml files from `deploy/` and `deploy/crds/` to
```
Comment (dantingl) [inline README.md:40]: later we should move crd files from deploy/crds/ to deploy/, NCP operator did that for certification

File: README.md [hunk 5]
```diff
@@ -1,2 +1,87 @@
...
+```
+
+At the moment the antrea-operator only works on Openshift 4
+environments
+
+## Documentation
+
+### Cluster network config
+Cluster network config is initially set in install-config.yaml, user could apply
+`Network.config.openshift.io` CRD to update `clusterNetwork` in `manifests/cluster-network-02-config.yml`.
```
Comment (dantingl) [inline README.md]: How about service CIDR

File: README.md [hunk 6]
```diff
@@ -1,2 +1,87 @@
...
+*Example configurations*
+```
+apiVersion: config.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  clusterNetwork:
+  - cidr: 10.10.0.0/14
+  networkType: ncp
```
Comment (dantingl) [inline README.md]: ncp -> antrea

File: README.md [hunk 7]
```diff
@@ -1,2 +1,87 @@
...
+```
+
+### Operator CR
+Operator CR `antrea-install` is used to provide antrea configurations:
+- AntreaAgentConfig holds the configurations for antrea-agent.
+- AntreaCNIConfig holds the configuration of CNI.
```
Comment (dantingl) [inline README.md]: configuration -> configurations

File: README.md (hunk 1 above)
Comment (ruicao93) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (ruicao93) [reply README.md]: Currently we don't support change `DisableMultiNetwork` option.

I checked the codes of project `openshift/cluster-network-operator`, the change `DisableMultiNetwork` option is not handled. I'm not sure if we need to support such case. I create an issue https://github.com/ruicao93/antrea-operator/issues/14 to track the issue.

File: README.md (hunk 3 above)
Comment (ruicao93) [reply README.md]: Sure, we could add `serviceNetwork` here as well. Done.

File: README.md (hunk 4 above)
Comment (ruicao93) [reply README.md:40]: Sure, let's track it in https://github.com/ruicao93/antrea-operator/issues/15

File: README.md (hunk 5 above)
Comment (ruicao93) [reply README.md]: Done. Add `serviceNetwork`.

File: README.md (hunk 6 above)
Comment (ruicao93) [reply README.md]: Done.

File: README.md (hunk 7 above)
Comment (ruicao93) [reply README.md]: Done

**Review by dantingl — APPROVED** [review]

**Pull request merged**